### DISCUSSION TITLE
Audiokit v2.2

### DISCUSSION THREAD

**Discussion by NaufalAzel** [pschatzmann/arduino-audiokit#87] [Q&A] [answered]
hi everyone
i have ai thinker esp32 audio kit v2.2 for my final project. what i want to ask is, is that possible to make ai thinker esp32 audio kit v2.2 get noise data as (db)

i try this one but what i got in serial monitor seems unrelevan
`#include "AudioKitHAL.h"
#include <math.h>

AudioKit kit;
const int BUFFER_SIZE = 1024;
uint8_t buffer[BUFFER_SIZE];

float calculateDB(uint8_t* buffer, size_t len) {
  float peakAmplitude = 0.0;
  for (size_t i = 0; i < len; i++) {
    float sample = (float)buffer[i] / 255.0; 
    float amplitude = fabs(sample);
    if (amplitude > peakAmplitude) {
      peakAmplitude = amplitude;
    }
  }
  float db = 20.0 * log10(peakAmplitude);
  return db;
}

void setup() {
  Serial.begin(115200);
  // Open in read mode
  auto cfg = kit.defaultConfig(AudioInputOutput);
  cfg.adc_input = AUDIO_HAL_ADC_INPUT_LINE1; 
  cfg.sample_rate = AUDIO_HAL_16K_SAMPLES;
  kit.begin(cfg);
  kit.setVolume(20);
}

void loop() {
  size_t len = kit.read(buffer, BUFFER_SIZE);

  
  float db = calculateDB(buffer, len);

  Serial.print("Noise Level (dB): ");
  Serial.println(db);
}
`
output : Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00
Noise Level (dB): 0.00

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you use the AudioTools API instead of this. There you can get the max amplitude with the help of the VolumeOutput class and you can apply your db conversion formula to this.

If you want to stick to this API, I still recommend that you read the AudioTools Introduction Wiki. There it is explained how to access the values: hint use uint8_t instead of int16_t values.

**Reply by NaufalAzel** [reply]
hello sir thank you so much for answering my question, but is that mean i cant code my audio kit v2.2 to get noise(dB) using this api? or i just need to use uint8_t instead of int16_t?

**Comment by pschatzmann** [comment]
You would obviously need to cast the data to the format that you request in the config! By default this is 16 bits per sample which would correspond to int16_t.  The AudioKit does not support 8 bits per sample!

**Reply by NaufalAzel** [reply]
so thats mean i still need use int16_t to get the input then convert it to int8_t

**Reply by pschatzmann** [reply]
Just read the AudioTools Introduction Wiki as I already told you! 
It is explained there with an example!
